app/controllers/registrations_controller.rb

```diff
@@ -1,17 +1,22 @@
 class RegistrationsController < ApplicationController
+  expose :company, -> { Company.find_by(subdomain: request.subdomain) }
   expose :user
 
   def new; end
 
   def create
-    session[:current_user_id] = user.id if user.save
-
-    respond_with user, location: root_path
+    session[:current_user_id] = register_user.user&.id
```

Comment:
Should we check that `register_user.success?` here?

Reply:
If user registrates, then he is not logged in, so `session[:current_user_id]` doesn't exist. And if `register_user` not success, we save `nil` to `session[:current_user_id]`. I don't see problem here and even more if-else construction ruins decent_exposure goodnesses.

Reply:
As I understand it, if either of the 2 interactors fails, the user will not know the reason. And the error will not be tracked on the server side.

Reply:
Oh... Yeah, you are right :relaxed: